### PR TITLE
boot, o/devicestate: split makeBootable20 into two parts 

### PR DESCRIPTION
**Pull request by anonymouse64**
Split makeBootable20 into two parts, one MakeRunnableSystem and
MakeBootableImage, with the former being exclusively for image-prepare time ops,
and the latter being for UC20 install mode preparing an active run system.

Additionally, for MakeRunnableSystem, remove out the final bit which sets the
boot variables and officially marks the point at which the system is ready to be
rebooted into the run mode. This will allow us to refactor the install mode
changes more easily in a future commit, but for now we get the same behavior
with EnsureNextBootToRunMode() in the install mode setup-run-system task.

Based on top of https://github.com/snapcore/snapd/pull/10018

### CODE REVIEW

File: boot/makebootable.go [hunk 1]
```diff
@@ -48,19 +48,20 @@ type BootableSet struct {
...
+// MakeBootableImage sets up the given bootable set and target filesystem
 // such that the system can be booted.
 //
-// rootdir points to an image filesystem (UC 16/18), image recovery
-// filesystem (UC20 at prepare-image time) or ephemeral system (UC20
-// install mode).
-func MakeBootable(model *asserts.Model, rootdir string, bootWith *BootableSet, sealer *TrustedAssetsInstallObserver) error {
+// rootdir points to an image filesystem (UC 16/18) or an image recovery
+// filesystem (UC20 at prepare-image time).
+// It should not point to an ephemeral system during UC20 install mode as
```
Comment (pedronis) [inline boot/makebootable.go:56]: this part of the comment is sligthly confusing,  naively one would expect to mention the actual run root system but is not what is used anyway as per my other comment, it should probably refer to the fact that on UC20 bootWithRecovery is expected to be true maybe?

File: boot/makebootable.go [hunk 2]
```diff
@@ -379,22 +386,9 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
...
-	}
-	blVars = map[string]string{
-		"snapd_recovery_mode": "run",
-	}
-	if err := bl.SetBootVars(blVars); err != nil {
-		return fmt.Errorf("cannot set recovery environment: %v", err)
-	}
+	// Note that we do not update the recovery bootloader env to actually
+	// transition to run mode here, that is left to the caller via something
+	// like boot.EnsureNextBootToRunMode()
```
Comment (pedronis) [inline boot/makebootable.go:391]: this kind of comment should also go in the doc comment for the function itself I think

**Review by pedronis — COMMENTED** [review]
last commit lgtm but some minor comments

File: boot/makebootable.go (hunk 1 above)
Comment (anonymouse64) [reply boot/makebootable.go]: sure I tried to rephrase this hopefully it makes sense

File: boot/makebootable.go [hunk 3]
```diff
@@ -227,7 +228,13 @@ func makeBootable20(model *asserts.Model, rootdir string, bootWith *BootableSet)
 	return nil
 }
 
-func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *BootableSet, sealer *TrustedAssetsInstallObserver) error {
+// MakeRunnableSystem is like MakeBootableImage in that it sets up a system to
+// be able to boot, but is unique in that it is intended to be called from UC20
+// install mode and makes the run system bootable (hence it is called "runnable")
+func MakeRunnableSystem(model *asserts.Model, rootdir string, bootWith *BootableSet, sealer *TrustedAssetsInstallObserver) error {
```
[the remark this answers is not in the document]
Comment (anonymouse64) [reply boot/makebootable.go]: done

File: boot/makebootable.go [hunk 4]
```diff
@@ -48,19 +48,20 @@ type BootableSet struct {
 	Recovery bool
 }
 
-// MakeBootable sets up the given bootable set and target filesystem
+// MakeBootableImage sets up the given bootable set and target filesystem
 // such that the system can be booted.
```
[the remark this answers is not in the document]
Comment (anonymouse64) [reply boot/makebootable.go]: done

File: boot/makebootable.go (hunk 2 above)
Comment (anonymouse64) [reply boot/makebootable.go]: moved this too (forgot initially, so I force pushed to add it to one of my other comments)

**Review by pedronis — APPROVED** [review]
thank you

**Review by bboozzoo — APPROVED** [review]
Thank you!

**Review by bboozzoo — APPROVED** [review]
Thank you!

**Pull request merged**